ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroDeserializerTest.java

```diff
@@ -105,4 +118,121 @@ public void shouldCreateCorrectRow() {
+    SchemaRegistryClient schemaRegistryClient = new MockSchemaRegistryClient();
+    List columns = Arrays.asList(1511897796092l, 1l, "item_1", 10.0, new Double[]{100.0},
+                                 Collections.singletonMap("key1", 100.0));
+
+    GenericRow genericRow = new GenericRow(columns);
+
+    KsqlGenericRowAvroDeserializer ksqlGenericRowAvroDeserializer = new
+        KsqlGenericRowAvroDeserializer(schema, schemaRegistryClient, false);
+
+    byte[] serializedRow = getSerializedRow("t1", schemaRegistryClient, avroSchema, genericRow);
+
+    GenericRow row = ksqlGenericRowAvroDeserializer.deserialize("t1", serializedRow);
+    Assert.assertNotNull(row);
+    assertThat("Incorrect deserializarion", row.getColumns().size(), equalTo(6));
+    assertThat("Incorrect deserializarion", (Long)row.getColumns().get(0), equalTo(1511897796092l));
+    assertThat("Incorrect deserializarion", (Long)row.getColumns().get(1), equalTo
+        (1l));
+    assertThat("Incorrect deserializarion", (String)row.getColumns().get(2), equalTo
+        ( "item_1"));
+
+    assertThat("Incorrect deserializarion", ((Double[])row.getColumns().get(4)).length, equalTo
+        (1));
+    assertThat("Incorrect deserializarion", ((Map)row.getColumns().get(5)).size(), equalTo
+        (1));
+
+
+  }
+
+
+  @Test
+  public void shouldDeserializeIfThereAreRedundantFields() {
+    org.apache.kafka.connect.data.Schema newSchema = SchemaBuilder.struct()
+        .field("ordertime".toUpperCase(), org.apache.kafka.connect.data.Schema.INT64_SCHEMA)
+        .field("orderid".toUpperCase(), org.apache.kafka.connect.data.Schema.INT64_SCHEMA)
+        .field("itemid".toUpperCase(), org.apache.kafka.connect.data.Schema.STRING_SCHEMA)
+        .field("orderunits".toUpperCase(), org.apache.kafka.connect.data.Schema.FLOAT64_SCHEMA)
+        .build();
+    SchemaRegistryClient schemaRegistryClient = new MockSchemaRegistryClient();
+    List columns = Arrays.asList(1511897796092l, 1l, "item_1", 10.0, new Double[]{100.0},
+                                 Collections.singletonMap("key1", 100.0));
+
+    GenericRow genericRow = new GenericRow(columns);
+
+    KsqlGenericRowAvroDeserializer ksqlGenericRowAvroDeserializer = new
+        KsqlGenericRowAvroDeserializer(newSchema, schemaRegistryClient, false);
+
+    byte[] serializedRow = getSerializedRow("t1", schemaRegistryClient, avroSchema, genericRow);
+    GenericRow row = ksqlGenericRowAvroDeserializer.deserialize("t1", serializedRow);
+    Assert.assertNotNull(row);
+    assertThat("Incorrect deserializarion", row.getColumns().size(), equalTo(4));
+    assertThat("Incorrect deserializarion", (Long)row.getColumns().get(0), equalTo(1511897796092l));
+    assertThat("Incorrect deserializarion", (Long)row.getColumns().get(1), equalTo
+        (1l));
+    assertThat("Incorrect deserializarion", (String)row.getColumns().get(2), equalTo
+        ( "item_1"));
+  }
+
+
+  @Test
+  public void shouldDeserializeWithMissingFields() {
```

Comment:
I'm not convinced of this one. If there are missing fields, then that should mean that the schemas are incompatible - right? Seems like an error to me

Reply:
Same as above. Consider user uses `create stream` to define the schema and the message is missing a field. Or at the time of creating the stream the avro schema had a field and later the schema changed and now we don't have the field. This is the same behavior for JSON format.